packages/logs/src/domain/trackNetworkError.ts

```diff
@@ -24,6 +27,16 @@ export function trackNetworkError(configuration: LogsConfiguration, errorObserva
 
   function handleCompleteRequest(type: RequestType, request: XhrCompleteContext | FetchCompleteContext) {
     if (!configuration.isIntakeUrl(request.url) && (isRejected(request) || isServerError(request))) {
+      if ('xhr' in request) {
+        onResponseDataAvailable(computeXhrResponseData(request.xhr, configuration))
+      } else if (request.response) {
+        computeFetchResponseText(request.response, configuration, onResponseDataAvailable)
+      } else if (request.error) {
+        onResponseDataAvailable(computeFetchErrorText(request.error, configuration))
+      }
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏it could be nice to have a more symmetric API to ease the understanding of the flow, something like:

```suggestion
      if ('xhr' in request) {
        computeXhrResponseData(request.xhr, configuration, onResponseDataAvailable)
      } else if (request.response) {
        computeFetchResponseText(request.response, configuration, onResponseDataAvailable)
      } else if (request.error) {
       computeFetchErrorText(request.error, configuration, onResponseDataAvailable)
      }
```

or by handling promise in `onResponseDataAvailable`:

```suggestion
      onResponseDataAvailable(
        'xhr' in request 
        ? computeXhrResponseData(request.xhr, configuration) 
        : request.response 
          ? computeFetchResponseText(request.response, configuration) 
          : computeFetchErrorText(request.error, configuration)
      )
```